src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsEventListener.java

```diff
@@ -19,15 +19,15 @@ public void onEvent(Event event) {
       case LOGIN:
       case IMPERSONATE:
         PrometheusExporter.instance().recordUserLogin(event);
+        PrometheusExporter.instance().recordUserEvent(event);
         break;
-      case LOGIN_ERROR:
-      case IMPERSONATE_ERROR:
-        PrometheusExporter.instance().recordFailedLogin(event);
       case LOGOUT:
         PrometheusExporter.instance().recordUserLogout(event);
+        PrometheusExporter.instance().recordUserEvent(event);
+        break;
       default:
-        // Ignore other event types for now
-        return;
+        PrometheusExporter.instance().recordUserEvent(event);
```

Comment:
default meaning exactly what ?

Reply:
In that case default means: take every event and count it. Only Login/Logout are treated special.

Reply:
I think this could go as a comment `// record all other events generically` or something.

Reply:
If a comment was needed, then we should rename the method instead IMO

Reply:
@josemigallas I think the method name describes exactly what it does: it record a user event. What's worth pointing out is why some events are treated different than others. I'll add comments for that.

Reply:
@josemigallas the `default` in that `switch` can not be changed. A comment on the _why_ is usually very helpful, IMO ; I agree w/ @pb82 here

Reply:
@matzew @pb82 
Sorry I wrote my comment only replying to Paolo's, I didn't really see the context, so:
I see your points, however I think it got unnecessarily verbose now. The code indeed explains itself well enough. See what I mean:
```java
  PrometheusExporter.instance().recordUserEvent(event);

  switch (event.getType()) {
      case LOGIN:
      case IMPERSONATE:
        PrometheusExporter.instance().recordUserLogin(event);
        break;
      case LOGOUT:
        PrometheusExporter.instance().recordUserLogout(event);
        break;
  }
```
The comments are redundant, login and impersonate both increase the logged in users counter, that exactly what the switch does.
On the other hand, the register event method call is independent from the switch so again, no need for comments.

What I think we should explain nevertheless is _why_ we consider impersonate as a login event.

Reply:
+1

Reply:
Suggestion sent as PR #2